internal/indexer/indexer.go

```diff
@@ -345,14 +398,16 @@ func (i *Indexer) indexDefinitionsForPackage(p *packages.Package) {
 	// implicit object for each case clause of a type switch (including default), and they all
 	// share the same position. This creates a map with one arbitrarily chosen argument for
 	// each distinct type switch.
-	caseClauses := map[token.Pos]types.Object{}
+	caseClauses := map[token.Pos]ObjectLike{}
 	for node, obj := range p.TypesInfo.Implicits {
 		if _, ok := node.(*ast.CaseClause); ok {
 			caseClauses[obj.Pos()] = obj
 		}
 	}
 
-	for ident, obj := range p.TypesInfo.Defs {
+	for ident, typeObj := range p.TypesInfo.Defs {
+		var obj ObjectLike = typeObj
```

Comment:
Why is this cast necessary?

Reply:
If I don't typecast, then we get this problem when we try and set obj:

```
1. cannot use caseClause (variable of type ObjectLike) as types.Object value in assignment: missing method color
```

Reply:
Oh it has those weird unexported methods on the interface. Gotcha.